controller/index.js

```diff
@@ -1,1019 +1,1054 @@
-    // if > 24 hours since; clear cache and wipe files 
-    // else move on
-    var table_key = ['agol', req.params.item, (req.params.layer || 0)].join(':');
-    agol.getInfo(table_key, function(err, info){
+    // returns data in the data
+    agol.getInfo(tableKey, function (err, info) {
+      if (err) {
+        agol.log('error', err)
+      }
 
-      // parse the spatial ref if we have one, 
+      // parse the spatial ref if we have one,
       // if its whitelisted remove it from the query object
       if (req.query.outSR) {
-        var sr = agol.parseSpatialReference(req.query.outSR);
-        if (sr && sr.wkid && [3785, 3857, 4326, 102100].indexOf(sr.wkid) !== -1){
-          delete req.query.outSR;
+        var sr = agol.parseSpatialReference(req.query.outSR)
+        if (sr && sr.wkid && [3785, 3857, 4326, 102100].indexOf(sr.wkid) !== -1) {
+          delete req.query.outSR
         }
       }
 
-      // sort the req.query before we hash so we are consistent 
-      var sorted_query = {};
-      _(req.query).keys().sort().each(function (key) {
-        if (key != 'url_only' && key != 'format'){
-          sorted_query[key] = req.query[key];
-        }
-      });
+      var key = controller._createCacheKey(req.params, req.query)
+      req.params.key = key
 
       // determine if this request is for a filtered dataset
-      req.query.isFiltered = (req.query.where || req.query.geometry) ? true : false;
-
-      // build the file key as an MD5 hash that's a join on the paams and look for the file 
-      var toHash = req.params.item + '_' + ( req.params.layer || 0 ) + JSON.stringify( sorted_query );
-      var key = crypto.createHash('md5').update(toHash).digest('hex');
-
-      var _returnProcessing = function( ){
-        if (typeof req.params.silent === 'undefined') {
-          agol.getCount(table_key, {}, function(err, count){
-            var code = 202;
-
-            // we need some logic around handling long standing processing times
-            var processingTime = ( Date.now() - info.retrieved_at)/1000 || 0;
-
-            var response = {
-              status: 'processing',
-              processing_time: processingTime, 
-              count: count
-            };
-            if ( info.generating ){
-              response.generating = info.generating;
-              // we received an error from the server
-              if (info.generating.error){
-                code = 502;
-              }
-            }
-            
-            agol.log('debug',JSON.stringify({status: code, item: req.params.item, layer: ( req.params.layer || 0 )})); 
-            res.status( code ).json( response );
-          });
+      req.query.isFiltered = (req.query.where || req.query.geometry)
+
+      // if the status is processing we either return with a file or a 202
+      if (info && info.status === 'processing') {
+        if (req.params.format) {
+          return controller._returnProcessingFile(req, res, info)
```

Comment:
This is really weird that we have a return processing for files and one for regular stuff. Even weirder that we start making a file inside a function called returnProcessing